src/box/raft.c

```diff
@@ -566,6 +573,7 @@ box_raft_try_promote(void)
 	}
 
 	raft_restore(raft);
```

Comment:
Can you consider moving `raft_restore` higher? The alternative solution would be adding assertions to the `raft_schedule_broadcast`. I wanna be sure, that tomorrow we won't introduce new `raft_on_update` triggers, which don't depend on `box_raft_try_promote`.  This will break it all again and reincarnate that bug